docs(core): improve caching feature page (#27173)
nrwl/nx: docs/shared/features/cache-task-results.md

# Cache Task Results

Rebuilding and retesting the same code repeatedly is costly. Nx offers a sophisticated and battle-tested computation caching system that ensures **code is never rebuilt twice**. This:

> - drastically **speeds up your task execution times** while developing locally, particularly [on CI](/ci/features/remote-cache)

Reviewer: ```suggestion
- drastically **speeds up your task execution times** while developing locally and even more [in CI](/ci/features/remote-cache)
```

- **saves you money on CI/CD costs** by reducing the number of tasks that need to be executed

> Nx **restores the terminal output, along with the files and artifacts** created from running the task (e.g., your build or dist directory). If you want to learn more about the conceptual model behind Nx's caching, read [How Caching Works](/concepts/how-caching-works).

Reviewer: ```suggestion
Nx **restores both the terminal output and the files** created from running the task (e.g., your build or dist directory). If you want to learn more about the conceptual model behind Nx's caching, read [How Caching Works](/concepts/how-caching-works).
```

the result of the test run.
{% /callout %}

## Enable Remote Caching

> By default, Nx caches task results locally. Most of the benefits from caching come on CI, where you can **share the cache between different runs**. To enable remote caching connect your workspace to [Nx Cloud](/nx-cloud) by running the following command:

Reviewer: ```suggestion
By default, Nx caches task results locally. The biggest benefit of caching comes from using remote caching in CI, where you can **share the cache between different runs**. To enable remote caching, connect your workspace to [Nx Cloud](/nx-cloud) by running the following command:
```

}
```

{% /tab %}
{% tab label="Project Level (package.json)" %}

```json {% fileName="packages/some-project/package.json"  %}
{
  "name": "some-project",
  "nx": {
    "targets": {
      "build": {
        ...
        "inputs": ["!{projectRoot}/**/*.md"],
        "outputs": ["{workspaceRoot}/dist/apps/some-project"],
        ...
      }
      ...
    }
  }
}
```

{% /tab %}
{% /tabs %}

Note that you only need to define output locations if they differ from the usual `dist` or `build` directory, which Nx automatically recognizes.

Learn more [about configuring inputs including `namedInputs`](/recipes/running-tasks/configure-inputs).

> ## Configure caching automatically

Reviewer: ```suggestion
## Configure Caching Automatically
```

> When using [Nx plugins](/concepts/nx-plugins), many caching configurations are handled automatically, saving you the effort of manual setup. **Nx plugins can [automatically infer tasks](/concepts/inferred-tasks) and configure caching** based on your underlying tooling configuration files.

Reviewer: ```suggestion
When using [Nx plugins](/concepts/nx-plugins), many tasks have caching configured automatically, saving you the effort of manual setup. **Nx plugins can [automatically infer tasks](/concepts/inferred-tasks) and configure caching** based on your underlying tooling configuration files.
```

For example, if you add the `@nx/vite` plugin using the following command...

```shell
npx nx add @nx/vite
```

...it automatically detects your `vite.config.ts` file, infers the tasks you'd be able to run, such as `build`, and **automatically configures the cache settings** for these tasks as well as the [task pipeline](/concepts/task-pipeline-configuration) (e.g., triggering dependent builds).

> This means **you don't need to manually specify cacheable operations for Vite tasks** and the cache setting such as input and outputs are always in sync with the `vite.config.ts` file.

Reviewer: ```suggestion
This means **you don't need to manually specify cacheable operations for Vite tasks** and the cache setting such as inputs and outputs are always in sync with the `vite.config.ts` file.
```

To view the task settings that have been automatically configured by a plugin, use the following command:

```shell
nx show project <project-name> --web
```

Alternatively, you can view these directly in your editor by installing [Nx Console](/getting-started/editor-setup).

Learn more details about [Nx plugins](/concepts/nx-plugins) and [inferred tasks](/concepts/inferred-tasks).

> ## Troubleshooting Caching

Reviewer: ```suggestion
## Troubleshoot Cache Settings
```
